File: baekjoon/p11659/Main.java
package baekjoon.p11659;

import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int m = sc.nextInt();

        int[] a = new int[n];
        for (int i = 0; i < a.length; i++) {
            a[i] = sc.nextInt();
        }

        int[][] b = new int[m][2];
        for (int i = 0; i < b.length; i++) {
            b[i][0] = sc.nextInt();
            b[i][1] = sc.nextInt();
        }

        for (int solve : solve(a, b)) {
            System.out.println(solve);
        }
    }

    private static int[] solve(int[] a, int[][] b) {
        int n = a.length;
        int[] sum = new int[n + 1];
        for (int i = 1; i <= n; i++) {
            sum[i] = sum[i - 1] + a[i - 1];
        }

        int[] ret = new int[b.length];
        for (int i = 0; i < ret.length; i++) {
            int start = b[i][0];
            int end = b[i][1];
            ret[i] = sum[end] - sum[start - 1];
        }

        return ret;
    }

}

File: baekjoon/p11659/Main.kt
package baekjoon.p11659

fun main() {
    val (n, m) = readLine()!!.split(" ").map { it.toInt() }
    val a = readLine()!!.split(" ").map { it.toInt() }
    val b = List(m) { readLine()!!.split(" ").map { it.toInt() } }

    for (solve in solve(a, b)) {
        println(solve)
    }
}

private fun solve(a: List<Int>, b: List<List<Int>>): List<Int> {
    val n = a.size
    val sum = IntArray(n + 1)
    for (i in 1..n) {
        sum[i] = sum[i - 1] + a[i - 1]
    }

    return (0 until b.size).map {
        val start = b[it][0]
        val end = b[it][1]
        sum[end] - sum[start - 1]
    }
}